lib/Conversion/TorchOnnxToTorch/DefaultDomainGtoP.cpp

```diff
@@ -1136,13 +1133,48 @@ void mlir::torch::onnx_c::populateDefaultDomainGtoP(
           return rewriter.notifyMatchFailure(binder.op,
                                              "dilations bind failure");
 
+        // set default padding
         if (padding.empty())
           padding.resize(spatial, 0);
         if (strides.empty())
           strides.resize(spatial, 1);
         if (dilations.empty())
           dilations.resize(spatial, 1);
 
+        if (padding.size() != static_cast<size_t>(spatial) &&
+            padding.size() != static_cast<size_t>(2 * spatial)) {
+          return rewriter.notifyMatchFailure(
+              binder.op, "padding list size does not match the number of axes");
+        }
```

Comment:
Hi @sriram-siloai, the overall patch looks good to me. I think we don't need this condition since, the `padding` tensor will satisfy this constraint no matter what the code flow is. I think these changes were adopted by the `Conv` op lowering and the case there's different, also to me it seemed a conservative check not a necessary check. Please remove this code.